xlunch.c

```diff
@@ -1078,33 +1083,33 @@ void run_command(char * cmd_orig)
     if (dont_quit)
     {
         pid_t pid=fork();
-        if (pid==0) // child process
-        {
+
+        switch (pid) {
+        case 0:   /* Child process */
             if (!multiple_instances) close(lock);
             printf("Forking command: %s\n",cmd);
-            int err;
-            err = execvp(array[0],array);
-            fprintf(stderr,"Error forking %s : %d\n",cmd,err);
-            exit(OKAY);
-        }
-        else if (pid<0) // error forking
-        {
-            fprintf(stderr,"Error running %s\n",cmd);
-        }
-        else // parent process
-        {
+            break;
+
+        case -1:  /* Error */
+            perror("fork");
+            /*FALLTHROUGH*/
+
+        default:  /* Parent */
             reset_prompt();
+            if (cmd != cmd_orig)
+                free(cmd);
+            return;
         }
     }
     else
     {
         cleanup();
         printf("Running command: %s\n",cmd);
-        int err;
-        err = execvp(array[0],array);
-        fprintf(stderr,"Error running %s : %d\n",cmd, err);
-        exit(OKAY);
     }
+
+    execvp(array[0],array);
+    fprintf(stderr,"Error running \"%s\": %s\n", array[0], strerror(errno));
```

Comment:
I  think this should still print out the full command, in the case where xlunch is used for scripting you might have multiple entries for the same program with different flags, and it might be interesting to see which one failed. Sorry for not spotting this earlier.

Reply:
No worries, this is accomplished.